src/plugins/messenger/MessengerPreview/components/MessengerGenericTemplate/MessengerGenericTemplate.tsx

```diff
@@ -107,26 +124,58 @@ export const getMessengerGenericTemplate = ({
                     firstButton?.focus();
                 }, 200);
             }             
+		}
+		
+        // Change the selectedItem state, in order to scroll the card with a focused element into view
+        handleScrollToView = (index) => {
+            this.setState({selectedItem: index})
         }
 
-        handleScrollToView = (index) => {
-            const focusedButton = document.activeElement;
-            focusedButton?.scrollIntoView({ behavior: "smooth", inline: "center", block: "nearest"});
+        handleCardChange = (index) => {
+            this.setState({selectedItem: index});
+        }
+
+        /**
+         * Move cards in carousel using arrow keys. 'useKeyboardArrows' prop of react-responsive-carousel
+         * moves the carousel irrespective of it has a focused item or not. Therefore, it has to be handled this way.
+         * 
+         */
+        handleArrowKeyDown = (event) => {
+            if(event.key === "ArrowRight") {
+                this.setState({selectedItem: this.state.selectedItem + 1}, () => {
+                    this.focusCardInView();
+                });
+            } else if(event.key === "ArrowLeft") {
+                this.setState({selectedItem: this.state.selectedItem - 1}, () => {
+                    this.focusCardInView();
+                })
+            }
+        }
+
+        focusCardInView = () => {
+            const nextCardToFocus = document.getElementById(`${this.carouselRootId}-${this.state.selectedItem}`);
```

Comment:
either use refs or a scoped element search

Reply:
I don't think it will be an issue here, as the id is a random id generated using uuid. Moreover, this part does not have any effect now due to the commented out codes